frontend/src/views/CellGuide/components/common/OntologyDagView/index.tsx

```diff
@@ -209,69 +258,42 @@ export default function OntologyDagView({
   const data = useMemo(() => {
     if (!treeData) return null;
     return hierarchy(treeData, (d) => {
-      const newChildren: TreeNodeWithState[] = [];
-      if (d.isExpanded) {
+      if (d.isExpanded && d.children && initialTreeState) {
```

Comment:
changes in this hook are for reducing cognitive complexity. i was using redundant conditionals which made the code hard to read.